pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -20,57 +20,19 @@ import (
 	"fmt"
 
 	"github.com/Azure/azure-sdk-for-go/arm/storage"
-	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
 const (
-	defaultStorageAccountType  = string(storage.StandardLRS)
-	fileShareAccountNamePrefix = "f"
+	defaultStorageAccountType      = string(storage.StandardLRS)
+	fileShareAccountNamePrefix     = "f"
+	sharedDiskAccountNamePrefix    = "ds"
+	dedicatedDiskAccountNamePrefix = "dd"
 )
 
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(shareName, accountName, accountType, location string, requestGiB int) (string, string, error) {
-	if len(accountName) == 0 {
-		// find a storage account that matches accountType
-		accounts, err := az.getStorageAccounts(accountType, location)
-		if err != nil {
-			return "", "", fmt.Errorf("could not list storage accounts for account type %s: %v", accountType, err)
-		}
-
-		if len(accounts) > 0 {
-			accountName = accounts[0].Name
-			glog.V(4).Infof("found a matching account %s type %s location %s", accounts[0].Name, accounts[0].StorageType, accounts[0].Location)
-		}
-
-		if len(accountName) == 0 {
-			// not found a matching account, now create a new account in current resource group
-			accountName = generateStorageAccountName(fileShareAccountNamePrefix)
-			if location == "" {
-				location = az.Location
-			}
-			if accountType == "" {
-				accountType = defaultStorageAccountType
-			}
-
-			glog.V(2).Infof("azureFile - no matching account found, begin to create a new account %s in resource group %s, location: %s, accountType: %s",
-				accountName, az.ResourceGroup, location, accountType)
-			cp := storage.AccountCreateParameters{
-				Sku:      &storage.Sku{Name: storage.SkuName(accountType)},
-				Tags:     &map[string]*string{"created-by": to.StringPtr("azure-file")},
-				Location: &location}
-			cancel := make(chan struct{})
-
-			_, errchan := az.StorageAccountClient.Create(az.ResourceGroup, accountName, cp, cancel)
-			err := <-errchan
-			if err != nil {
-				return "", "", fmt.Errorf(fmt.Sprintf("Failed to create storage account %s, error: %s", accountName, err))
-			}
-		}
-	}
-
-	// find the access key with this account
-	accountKey, err := az.getStorageAccesskey(accountName)
+	accountName, accountKey, err := az.SearchStorageAccount(accountName, accountType, location, fileShareAccountNamePrefix)
```

Comment:
We'd better not assign value to 'accountName' here.

Consider the code in SearchStorageAccount:
```
+	accountKey, err := az.getStorageAccesskey(accountName)
 +	if err != nil {
 +		return "", "", fmt.Errorf("could not get storage key for storage account %s: %v", accountName, err)
 +	}
```

If user specifies 'AccountName', but later getStorageAccesskey fails, then SearchStorageAccount returns empty for first parameter.

In this case the follow-up error reported would use empty value for accountName, which does not sound good to user.

Also for CreateVolume.

Reply:
good catch, fixed both funcs.